waspc/data/Generator/templates/server/src/jobs/core/pgBoss/pgBossJob.js

```diff
@@ -68,17 +69,28 @@ class PgBossJob extends Job {
  * functions, we will override the previous calls.
  * @param {string} jobName - The user-defined job name in their .wasp file.
  * @param {fn} jobFn - The user-defined async job callback function.
- * @param {object} defaultJobOptions - PgBoss specific options for boss.send() applied to every submit() invocation,
+ * @param {object} defaultJobOptions - pg-boss specific options for boss.send() applied to every submit() invocation,
  *                                     which can overriden in that call.
+ * @param {object} jobSchedule [Optional] - The 5 field cron string, job function JSON arg, and `boss.send()` options when invoking the job.
  */
-export async function createJob({ jobName, jobFn, defaultJobOptions } = {}) {
-  // As a safety precaution against undefined behavior of registering different
-  // functions for the same job name, remove all registered functions first.
-  await boss.offWork(jobName)
+export function createJob({ jobName, jobFn, defaultJobOptions, jobSchedule } = {}) {
+  pgBossStarted.then(async (boss) => {
+    // As a safety precaution against undefined behavior of registering different
+    // functions for the same job name, remove all registered functions first.
+    await boss.offWork(jobName)
 
-  // This tells pgBoss to run given worker function when job/payload with given job name is submitted.
-  // Ref: https://github.com/timgit/pg-boss/blob/master/docs/readme.md#work
-  await boss.work(jobName, jobFn)
+    // This tells pgBoss to run given worker function when job/payload with given job name is submitted.
+    // Ref: https://github.com/timgit/pg-boss/blob/master/docs/readme.md#work
+    await boss.work(jobName, jobFn)
+
+    // If a job schedule is provided, we should schedule the recurring job.
+    // If the schedule name already exists, it's updated to the provided cron expression, arguments, and options.
+    // Ref: https://github.com/timgit/pg-boss/blob/master/docs/readme.md#scheduling
+    if (jobSchedule) {
+      const options = { ...defaultJobOptions, ...jobSchedule.options }
+      await boss.schedule(jobName, jobSchedule.cron, jobSchedule.args || null, options)
+    }
+  })
```

Comment:
So this piece is not awaited for, instead it happens in parallel / in the background.
This looks like a problem to me since people can try to submit new job but `boss.work` might not have finished yet! So I would instead do `const boss = await pgBossStarted` and then the rest of the logic.

Reply:
We chatted and this is the async at module level problem. It is saved by pg-boss allowing you to submit jobs even before you register the worker and it still works, even if the awaits are not resolved in FIFO order. I will add a comment here.